### ISSUE TITLE
undefined method

### ISSUE THREAD

**Issue by confact** [DouweM/shoutout#1]
Hi!

I am using your package for a hobby project. I get `undefined method '[]' for nil:NilClass` when trying to get a stream that don't give a HTTP status, It's either ICY status or no status at all. Is it some way to fix this?

I have tried to fix it, with a ugly code in 42 in stream.rb:

```
begin
    status_code = status_line.match(/\AICY ([0-9]{3})/)[1].to_i
rescue NoMethodError
    begin
        status_code = status_line.match(/\AHTTP\/([0-9]\.[0-9]) ([0-9]{3})/)[2].to_i
    rescue NoMethodError
        status_code = 200 # for streams that don't give icy headers nor http headers
    end
end
```

still doesn't work good as I want it too. It hangs or just `undefined method '[]' for nil:NilClass`

How to fix this issue?

**Comment by confact** [comment]
A example how it can look like:

```
# curl -H "icy-metadata: 1" -XGET 'http://s10.voscast.com:7950'
ICY 200 OK
icy-notice1:<BR>This stream requires <a href="http://www.winamp.com/">Winamp</a><BR>
icy-notice2:SHOUTcast Distributed Network Audio Server/posix(linux x64) v2.0.0.29<BR>
icy-name:Star KSA
icy-genre:Top40
icy-url:www.ananaz.com/StarKSA
content-type:audio/mpeg
icy-pub:1
icy-metaint:8192
icy-br:128
```

**Comment by DouweM** [comment]
Interesting, that SHOUTcast server speaks a dialect I hadn't encountered before :) It claims to speak "ICY" rather than "HTTP" in its status line, and it has `icy-notice1` and `icy-notice2` headers rather than a singular `icy-notice`.

I've fixed the issue like so: 3741c61cb1445765df02881e3812bc7b63194c5f
